### PR TITLE
BaSP-TG-08: Tasks table resources

### PR DESCRIPTION
**Pull request by DinoStampella**

### CODE REVIEW

**Review by kolodani — APPROVED** [review]
great work, well done

File: src/Components/Tasks/index.js [hunk 1]
```diff
@@ -1,11 +1,119 @@
...
+      setTimeout(() => setSuccessMessage(''), 2000);
+      hideModifyModal();
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  //Add task
+  const addTask = async (data) => {
+    console.log(JSON.stringify(data));
```
Comment (Alfondalix) [inline src/Components/Tasks/index.js:76]: We should delete this line

File: src/Components/Tasks/index.js [hunk 2]
```diff
@@ -1,11 +1,119 @@
...
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/tasks`);
+      const data = await response.json();
+      setTasksList(data.data);
+    } catch (err) {
+      console.log(err);
+    }
+  }, [modifyModalControl]);
+
+  // Delete Task
```
Comment (Alfondalix) [inline src/Components/Tasks/index.js:24]: We could delete this comment

File: src/Components/Tasks/index.js [hunk 3]
```diff
@@ -1,11 +1,119 @@
...
+      console.log(error);
+    }
+  };
+
+  // Show edit modal
+  const showModifyModal = (id) => {
+    setModifyModalControl({ id: id, modal: true });
+  };
+
+  // Hide edit modal
```
Comment (Alfondalix) [inline src/Components/Tasks/index.js:43]: Same as above

File: src/Components/Tasks/index.js [hunk 4]
```diff
@@ -1,11 +1,119 @@
...
+      }
+      setSuccessMessage('Task edited successfully' || res.message);
+      setTimeout(() => setSuccessMessage(''), 2000);
+      hideModifyModal();
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  //Add task
```
Comment (Alfondalix) [inline src/Components/Tasks/index.js:74]: Same here

File: src/Components/Tasks/index.js [hunk 5]
```diff
@@ -1,11 +1,119 @@
...
+        setErrorMessage(
+          res.message[0].message || res.message || 'An unexpected error has occurred'
+        );
+        return;
+      }
+      setSuccessMessage('Task added successfully' || res.message);
+      setTimeout(() => setSuccessMessage(''), 2000);
+      hideModifyModal();
+    } catch (error) {
+      console.log(error);
```
Comment (Alfondalix) [inline src/Components/Tasks/index.js:96]: We could use console.error(error) instead

File: src/Components/Tasks/index.js [hunk 6]
```diff
@@ -1,11 +1,119 @@
...
+  const [errorMessage, setErrorMessage] = useState('');
+  const [successMessage, setSuccessMessage] = useState('');
+
+  useEffect(async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/tasks`);
+      const data = await response.json();
+      setTasksList(data.data);
+    } catch (err) {
+      console.log(err);
```
Comment (Alfondalix) [inline src/Components/Tasks/index.js:20]: Same as the other comment about console.error

File: src/Components/Tasks/index.js [hunk 7]
```diff
@@ -1,11 +1,119 @@
...
+        setErrorMessage(
+          res.message[0].message || res.message || 'An unexpected error has occurred'
+        );
+        return;
+      }
+      setSuccessMessage('Task edited successfully' || res.message);
+      setTimeout(() => setSuccessMessage(''), 2000);
+      hideModifyModal();
+    } catch (error) {
+      console.log(error);
```
Comment (Alfondalix) [inline src/Components/Tasks/index.js:70]: SAme here

**Review by Alfondalix — CHANGES_REQUESTED** [review]

**Review by LuisQueff — APPROVED** [review]
Ready to merge

**Review by kolodani — APPROVED** [review]
nice work!!!

**Review by Alfondalix — APPROVED** [review]

**Pull request merged**